dataproc/quickstart/quickstart_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	dataproc "cloud.google.com/go/dataproc/apiv1"
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	dataprocpb "google.golang.org/genproto/googleapis/cloud/dataproc/v1"
+)
+
+// Underscored names so as to not collide with sample declarations
+var (
+	_clusterName string
+	bktName      string
+	_jobFilePath string
+	jobFName     = "sum.py"
+	code         = `import pyspark
+sc = pyspark.SparkContext()
+rdd = sc.parallelize((1,2,3,4,5))
+sum = rdd.reduce(lambda x, y: x + y)`
+	_region = "us-central1"
+)
+
+func setup(t *testing.T, tc testutil.Context) {
+	ctx := context.Background()
+	flag.Parse()
+
+	_clusterName = "go-qs-test-" + tc.ProjectID
+	bktName = "go-dataproc-qs-test-" + tc.ProjectID
+	_jobFilePath = fmt.Sprintf("gs://%s/%s", bktName, jobFName)
+
+	sc, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Errorf("Error creating storage client with error: %v", err)
+	}
+
+	bkt := sc.Bucket(bktName)
+	if err := bkt.Create(ctx, tc.ProjectID, nil); err != nil {
```

Comment:
Tests get messed up and buckets don't get deleted. Please copy and paste https://github.com/GoogleCloudPlatform/golang-samples/blob/master/storage/objects/main_test.go#L363. (I'll move that to a shared location in a future PR.)